### PR TITLE
This is my second proposal

### PR DESCRIPTION
**Pull request by DanielSmedegaardBuus**
This one uses a single "options" hash as 2nd parameter in the test case function.

It would make it easier to extend stuff later without breaking stuff backwards, but if you accept it, it means you'd have to change your loadbooster code to match...

Hope you accept it :D

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -10,6 +10,36 @@ xmlParser.parseString(fs.readFileSync(__dirname+"/iedoc-core.xml","utf-8"), func
...
+      fs.mkdirSync(path);
+    } else if (!fs.statSync(path).isDirectory()) {
+        throw new Error("Cannot create directory '" + path + "'. File of same name already exists.");
+    }
+  }
+}
+
+exports.setLogger=function (logger){
+    var formatter = require(__dirname+"/JavascriptFormatter");
+    formatter.setLogger(logger);
```
Comment (flyingfisher) [inline index.js:39]: The logger here is not checked. The logger may miss error or warn implementation, lead to undefined error.
I suggest to use lodash, `_.defaults`.

```
_.defaults(logger,{
     log   : function () {},
     debug : function () {},
     info  : function () {},
     warn  : console.warn,
     error : console.error
})
```

**Comment by flyingfisher** [comment]
Thank you very much, for what you've done.
I've read my loadbooster code. The package version is fixed. It will only take effect when I use update command.
So I take your second proposal.

**Comment by flyingfisher** [comment]
I've added lodash :-)

**Comment by DanielSmedegaardBuus** [comment]
Cool! :D Thanks a lot :+1:

**Pull request merged**